h/views/api/auth.py

```diff
@@ -205,10 +206,11 @@ def post(self):
             self.request.POST,
             self.request.headers,
         )
+
         if status == 200:
             return json.loads(body)
 
-        raise exception_response(status, body=body)
+        raise exception_response(status, detail=_decode_json_if_valid(body))
```

Comment:
The existing solution appeared to not decode on purpose. This makes me think this can be a string or other non decodeable things, so I've replicated that, but decoded when we can. Otherwise we get errors like:

```{..., "details": "{\"stuff\": \"etc."}"}```

Reply:
The client expects the `reason` field to be a string, for all API responses. Other browser-based clients may make the same assumption.

Reply:
If we were implementing the OAuth spec strictly, the error responses should I think follow https://datatracker.ietf.org/doc/html/rfc6749#section-5.2. This would be a backwards incompatible change though so we'd need a migration plan, and that's out of scope here.